### PR TITLE
Thematics

### PR DESCRIPTION
**Pull request by gparant**

### CODE REVIEW

File: play/src/front/Phaser/Game/MapEditor/Tools/AreaEditorTool.ts [hunk 1]
```diff
@@ -436,6 +436,7 @@ export class AreaEditorTool extends MapEditorTool {
 
     public handleAreaDeletion(id: string, areaData: AreaData | undefined): void {
         this.scene.getGameMapFrontWrapper().listenAreaDeletion(areaData);
+        this.scene.getGameMapFrontWrapper().areasManager.removeArea(id);
```
Comment (moufmouf) [inline play/src/front/Phaser/Game/MapEditor/Tools/AreaEditorTool.ts:439]: 1. Is this normal that this method is not called in `TrashEditorTool.handleAreaDeletion`.
2. Shouldn't we move the call to `areasManager.removeArea` inside `listenAreaDeletion`?

File: play/src/front/Phaser/Game/MapEditor/Tools/AreaEditorTool.ts [hunk 2]
```diff
@@ -448,6 +449,7 @@ export class AreaEditorTool extends MapEditorTool {
 
     public handleAreaCreation(config: AreaData, localCommand: boolean): void {
         this.scene.getGameMapFrontWrapper().listenAreaCreation(config);
+        this.scene.getGameMapFrontWrapper().areasManager.addArea(config);
```
Comment (moufmouf) [inline play/src/front/Phaser/Game/MapEditor/Tools/AreaEditorTool.ts:452]: Same comment:

1. Is this normal that this method is not called in `TrashEditorTool.handleAreaCreation`.
2. Shouldn't we move the call to `areasManager.addArea` inside `listenAreaCreation`?

File: play/src/front/Phaser/Game/MapEditor/Tools/AreaEditorTool.ts [hunk 3]
```diff
@@ -463,6 +465,7 @@ export class AreaEditorTool extends MapEditorTool {
 
     public handleAreaUpdate(oldConfig: AtLeast<AreaData, "id">, newConfig: AtLeast<AreaData, "id">): void {
         this.scene.getGameMapFrontWrapper().listenAreaChanges(oldConfig, newConfig);
+        this.scene.getGameMapFrontWrapper().areasManager.updateArea(newConfig);
```
Comment (moufmouf) [inline play/src/front/Phaser/Game/MapEditor/Tools/AreaEditorTool.ts:468]: Shouldn't we move the call to areasManager.updateArea inside listenAreaChanges?

File: play/src/i18n/en-US/mapEditor.ts [hunk 4]
```diff
@@ -169,6 +169,16 @@ const mapEditor: BaseTranslation = {
...
         areaSerchable: "Searchable in the exploration mode",
         addDescriptionField: "Add description field",
+        addRight: "Add rights",
+        rightTitle: "Right by user tag (write / read)",
+        rightDescription:
+            "Rights define who can interact with the area. If you leave it empty, everyone can use it. If you set it, only users who have at least one of these 'tags' can use it.",
+        rightWriteTitle: "Write rights",
+        rightWriteDescription:
+            "Write rights define who can modify the area. The user who have at least on of these tags can create, update or delete an object in the area.",
+        rightReadTitle: "Read rights",
```
Comment (moufmouf) [inline play/src/i18n/en-US/mapEditor.ts:179]: I think we talked about renaming read/write to something else? Did we not?
Something like "edit right" and "enter / use right" ?

**Review by moufmouf — COMMENTED** [review]
I generally feel inconfortable with the name "Thematics" to describe a right management system.
This is the way the CEA sees it, but it does not have to be the way we name this feature.

I'd rather talk about "right restricted areas" or "restricted areas". We should remove any mention of the "Thematics" name and replace it with "RestrictedAreas" instead (or something else if you have a better idea)

For instance, instead of `isGameMapContainsThematics`, go for `isGameMapContainsRestrictedAreas` (or better: `doesGameMapContainsThematics`)

File: messages/protos/messages.proto [hunk 5]
```diff
@@ -218,6 +220,8 @@ message CreateEntityMessage {
   string prefabId = 5;
   repeated google.protobuf.Value properties = 6;
   optional string name = 7;
+  uint32 x_center = 8;
+  uint32 y_center = 9;
```
[the remark this answers is not in the document]
Comment (matthintosh) [reply messages/protos/messages.proto]: Fix by replacing center coordinates with width and height

File: play/src/i18n/en-US/mapEditor.ts (hunk 4 above)
Comment (matthintosh) [reply play/src/i18n/en-US/mapEditor.ts:179]: I don't think so. But I can modify it if you prefer.

File: play/src/front/Phaser/Game/GameMap/AreasManager.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
+import { AreaData, AreaPermissions, AtLeast, GameMapAreas } from "@workadventure/map-editor";
+import { Area } from "../../Entity/Area";
+import { GameScene } from "../GameScene";
+import { mapEditorActivatedForThematics } from "../../../Stores/MenuStore";
+
+/**
+ * This class handle the display
```
Comment (moufmouf) [inline play/src/front/Phaser/Game/GameMap/AreasManager.ts:7]:
```suggestion
 * This class handles the display
```

File: play/src/i18n/en-US/mapEditor.ts [hunk 7]
```diff
@@ -169,6 +169,16 @@ const mapEditor: BaseTranslation = {
         areaDescriptionPlaceholder: "My area is a...",
         areaSerchable: "Searchable in the exploration mode",
         addDescriptionField: "Add description field",
+        addRight: "Add rights",
+        rightTitle: "Right by user tag (write / read)",
```
Comment (moufmouf) [inline play/src/i18n/en-US/mapEditor.ts:173]:
```suggestion
        rightTitle: "Access / edition rights by user tag",
```

File: play/src/i18n/fr-FR/mapEditor.ts [hunk 8]
```diff
@@ -167,6 +167,14 @@ const mapEditor: DeepPartial<Translation["mapEditor"]> = {
         areaDescriptionPlaceholder: "Description de la zone",
         areaSerchable: "Recherchable dans le mode exploration",
         addDescriptionField: "Ajouter une description",
+        addRight: "Ajouter des droits",
+        rightTitle: "Droit via les tags utilisateur (écriture / lecture)",
```
Comment (moufmouf) [inline play/src/i18n/fr-FR/mapEditor.ts:171]:
```suggestion
        rightTitle: "Droit d'accès et d'édion via les tags utilisateur",
```

**Pull request merged**